bioio_ome_tiled_tiff/reader.py

```diff
@@ -44,66 +63,184 @@ class Reader(BaseReader):
+
+        # Currently do not support custom chunking, throw a warning.
+        if chunk_dims is not None:
+            log.warning(
+                "OmeTiledTiffReader does not currently support custom chunking."
+            )
 
     @property
     def scenes(self) -> Tuple[str, ...]:
-        """
-        Return the list of available scenes for the file using the
-        cached parameters stored to the object in the __init__.
-        """
-        raise NotImplementedError()
+        return tuple(image_meta.id for image_meta in self._rdr.metadata.images)
 
-    def _read_delayed(self) -> "xr.DataArray":
-        """
-        Return an xarray DataArray filled with a delayed dask array, coordinate planes,
-        and any metadata stored in the attrs.
+    @property
+    def current_scene(self) -> str:
+        return self.scenes[self._current_scene_index]
 
-        Metadata should be labelled with one of the bioio-base constants.
-        """
-        raise NotImplementedError()
+    @property
+    def current_scene_index(self) -> int:
+        return self._current_scene_index
 
-    def _read_immediate(self) -> "xr.DataArray":
+    def set_scene(self, scene_id: Union[str, int]) -> None:
         """
-        Return an xarray DataArray filled with an in-memory numpy ndarray,
-        coordinate planes, and any metadata stored in the attrs.
-
-        Metadata should be labelled with one of the bioio-base constants.
+        For all BfioReader subclasses, the only allowed value is the name of the first
+        scene since only the first scene can be read by BioReader objects. This method
+        exists primarily to help this Reader fit into existing unit test templates and
+        in case BioReader is updated to support multiple scenes.
+
+        Parameters
+        ----------
+        scene_id: Union[str, int]
+            The scene id (if string) or scene index (if integer)
+            to set as the operating scene.
+
+        Raises
+        ------
+        IndexError
+            The provided scene id or index does not reference the first scene.
+        TypeError
+            The provided value wasn't a string (scene id) or integer (scene index).
         """
-        raise NotImplementedError()
+        # Route to int or str setting
+        if isinstance(scene_id, (str, int)):
+            # Only need to run when the scene id is different from current scene
+            if scene_id not in (self.current_scene, self.current_scene_index):
+                raise IndexError(
+                    "Scene id: Cannot change scene for "
+                    + f"{self.__class__.__name__} objects."
+                )
+
+        else:
+            raise TypeError(
+                f"Must provide either a string (for scene id) "
+                f"or integer (for scene index). Provided: {scene_id} ({type(scene_id)}."
+            )
 
-    # Optional Methods
+    @property
+    def channel_names(self) -> Optional[List[str]]:
```

Comment:
I wonder why this was implemented. It is different than the super's matching property implementation. Doesn't seem like your problem for this PR though.